### ISSUE TITLE
Exclude completions of binding pattern variable initializers

### ISSUE THREAD

**Issue by zardoy** [microsoft/TypeScript#53049]
# Bug Report

Completions from binding pattern variable initializers are included, but they shouldn't

### 🔎 Search Terms

no suggestions const destruct destructure

### 🕗 Version & Regression Information

Copied from available fix https://github.com/microsoft/TypeScript/pull/52723 :

```ts
const {a1} = a/* should not complete a1 */
```

Real world case:

```ts
let pageToReturn

const { data: pageData } = await octokit.repos.listTags({
    page: page /* pageData is suggested first */
})
```

Some kind of follow up for https://github.com/microsoft/TypeScript/pull/42087

**Comment by RyanCavanaugh** [comment]
Care should be taken to not cause false negatives, e.g.
```ts
const { tick } = {
  tick() {
    window.setTimeout(tick, 100);
    //                ^^^^ valid
  }
}
```
